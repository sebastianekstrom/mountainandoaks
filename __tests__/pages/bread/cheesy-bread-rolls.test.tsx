import { render, screen } from "@testing-library/react";
import React from "react";

import CheesyBreadRolls from "pages/bread/cheesy-bread-rolls";

describe("Cheesy bread rolls page", () => {
	it("renders title correctly", () => {
		render(<CheesyBreadRolls />);
		const title = screen.queryAllByText(/Cheesy bread rolls/i);
		expect(title).toBeDefined();
	});

	it("renders the ingress text", () => {
		render(<CheesyBreadRolls />);
		const ingress = screen.queryAllByText(
			/Made with Manitoba flour, these rolls combine a chewy interior with a crispy/i,
		);
		expect(ingress).toBeDefined();
	});

	it("renders the method steps", () => {
		render(<CheesyBreadRolls />);
		const step = screen.queryAllByText(
			/Divide the dough into approximately 70-gram pieces/i,
		);
		expect(step).toBeDefined();
	});

	it("renders the correct ingredients", () => {
		render(<CheesyBreadRolls />);
		const ingredientList = screen.getByTestId("INGREDIENT_LIST");
		expect(ingredientList).toMatchSnapshot();
	});
});
