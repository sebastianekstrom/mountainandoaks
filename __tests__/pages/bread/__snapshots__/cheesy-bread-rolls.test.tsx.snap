// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheesy bread rolls page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Manitoba flour
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          260
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Water
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          146
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class="flex flex-row items-center"
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Egg
          </p>
          <span
            class="ml-2 flex items-center"
          >
            <button
              class="outline-none focus:shadow-[0_0_0_2px] focus:shadow-black"
              data-state="closed"
              type="button"
            >
              <img
                alt=""
                class="w-[15] h-auto"
                height="15"
                quality="100"
                src="/images/misc/question-mark.svg"
                width="15"
              />
            </button>
          </span>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          24
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Dry yeast
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          9
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Salt
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          6
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Sugar
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Butter
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Slices of cheese
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          6
        </p>
      </li>
    </ul>
    <div
      class="ml-auto"
    >
      <button
        class="mt-2 rounded-full text-regular bg-pill-background py-2 px-4"
        data-testid="BAKERS_PERCENTAGE"
        role="button"
        type="button"
      >
        <p
          class="variable font-sans text-caption leading-paragraph "
        >
          Show bakers %
        </p>
      </button>
    </div>
  </div>
</section>
`;
